gst/datarepo/gstdatareposrc.c

```diff
@@ -373,6 +392,54 @@ gst_data_repo_src_get_file_offset (GstDataRepoSrc * src, guint64 sample_index)
   return offset;
 }
 
+/**
+ * @brief Function to shuffle samples index
+ */
+static void
+gst_data_repo_src_shuffle_samples_index (GstDataRepoSrc * src)
+{
+  guint i, j;
+  gint value_i, value_j;
+  g_return_if_fail (src != NULL);
+  g_return_if_fail (src->shuffled_index_array != NULL);
+
+  GST_LOG_OBJECT (src, "samples index are shuffled");
+
+  /* Fisher-Yates algorithm */
+  /* The last index is the number of samples - 1. */
+  for (i = src->num_samples - 1; i > 0; i--) {
```

Comment:
Don't need to shuffle i=0?

Reply:
in each read function, it is checked below.
    if (src->is_shuffle)
      gst_data_repo_src_shuffle_samples_index (src);